inc/cleantalk-common.php

```diff
@@ -615,50 +607,86 @@ function delete_spam_comments() {
 * Get data from submit recursively
 */
 
-function ct_get_fields_any(&$email,&$message,&$nickname,&$subject, &$contact,$arr)
-{
+function ct_get_fields_any($arr, $message=array(), $email=NULL, $nickname=NULL, $subject=NULL, $contact=true){
+	
 	$skip_params = array(
 	    'ipn_track_id', // PayPal IPN #
 	    'txn_type', // PayPal transaction type
 	    'payment_status', // PayPal payment status
 	    'ccbill_ipn' //CCBill IPN 
     );
-   	foreach($skip_params as $key=>$value)
-   	{
-   		if(@array_key_exists($value,$_GET)||@array_key_exists($value,$_POST))
-   		{
+	
+    $obfuscate_params = array(
+        'password',
+        'password0',
+        'password1',
+        'password2',
+        'pass',
+        'pwd',
+        'user_pass'
+    );
+	
+   	foreach($skip_params as $key=>$value){
+   		if(@array_key_exists($value,$_GET)||@array_key_exists($value,$_POST)){
    			$contact = false;
    		}
    	}
-	foreach($arr as $key=>$value)
-	{
-		if(!is_array($value)&&!is_object($value)&&@get_class($value)!='WP_User')
-		{
-			if (in_array($key, $skip_params) && $key!=0 && $key!='' || preg_match("/^ct_checkjs/", $key)) {
+	
+	foreach($arr as $key=>$value){
+		if(!is_array($value)&&!is_object($value)&&@get_class($value)!='WP_User'){
+			if (in_array($key, $skip_params) && $key!=0 && $key!='' || preg_match("/^ct_checkjs/", $key)){
                 $contact = false;
             }
-			if ($email === '' && @preg_match("/^\S+@\S+\.\S+$/", $value))
-	    	{
+			if (!$email && @preg_match("/^\S+@\S+\.\S+$/", $value)){
 	            $email = $value;
 	        }
-	        else if ($nickname === '' && ct_get_data_from_submit($value, 'name'))
-	    	{
+	        else if (!$nickname && ct_get_data_from_submit($key, 'name') && !ct_get_data_from_submit($key, '_hp_name')){
 	            $nickname = $value;
 	        }
-	        else if ($subject === '' && ct_get_data_from_submit($value, 'subject'))
-	    	{
+	        else if (!$subject && ct_get_data_from_submit($key, 'subject')){
 	            $subject = $value;
+	        }else{
+                //
+                // Obfuscate private data
+                //
+                if (in_array($key, $obfuscate_params)) {
+                    $value = ct_obfuscate_param($value); 
+                }
+	        	$message[$key] = $value;
 	        }
-	        else
-	        {
-	        	@$message.="$value\n";
-	        }
-		}
-		else if(!is_object($value)&&@get_class($value)!='WP_User')
-		{
-			@ct_get_fields_any($email, $message, $nickname, $subject, $contact, $value);
+		}else if(!is_object($value)&&@get_class($value)!='WP_User'){
+			$temp=ct_get_fields_any($value, $message, $email, $nickname, $subject, $contact);
+			
+			$sender_email = ($temp['email'] ? $temp['email'] : '');
```

Comment:
Why this variable $sender_email is declared, but doesn't used in the function further? Same question for $sender_nickname,$subject,$contact_form,$message.